wrappers/src/list_cs.cpp

```diff
@@ -168,8 +168,8 @@ REALM_EXPORT size_t list_find_value(List& list, realm_value_t value, NativeExcep
                 throw ObjectManagedByAnotherRealmException("Can't look up index of an object that belongs to a different Realm.");
             }
 
-            if ((list_type & PropertyType::Flags) == PropertyType::Mixed) {
-                return list.find_any(ObjLink(value.link.object->get_object_schema().table_key, value.link.object->obj().get_key()));
+            if ((list_type & ~PropertyType::Flags) == PropertyType::Mixed) {
```

Comment:
I'm curious about this: this mistake made sure that any list holding Mixed values were never recognized. Ending up at line 175 where the find, instead, was called. Here it's where I get confused: was that working fine or throwing an error? I can see that List::Find relies on as<T>() which tries to statically convert the list to a list of Mixed in our case. If this conversion was possible then what's the point of find_any? Can I assume JUST performance?
If I have misread all this, then what did I miss?

Reply:
This is actually a good question. If we arrive at line 175 with a Mixed property the there's an exception coming from `Collection::validate(const Obj& obj)`, due to the fact that the obj type is not the same of the list type (due to mixed). There is actually even a `// FIXME: This does not work for TypedLink.` there, so I suppose this part here could be fixed, but I am not sure if there are other reasons for not using `find_any` all the time. Maybe it's just a matter of avoiding the conversion from `realm_value_t` to `Mixed`?
Besides, in Results, we do not have this difference between `index_of` and `index_of_any`, so it's a little confusing.
@nirinchev This is probably something we need to talk about :)

Reply:
Yeah, let's chat about it at the roundtable